x/evm/types/params.go

```diff
@@ -51,6 +52,20 @@ func (p Params) Validate(ac address.Codec) error {
 		return ErrInvalidNumRetainBlockHashes
 	}
 
+	if p.GasEnforcement != nil {
+		if p.GasEnforcement.MaxGasFeeCap != nil && p.GasEnforcement.MaxGasFeeCap.IsNegative() {
+			return ErrInvalidGasEnforcement
+		}
+		
+		for i, addrStr := range p.GasEnforcement.UnlimitedGasSenders {
+			contractAddr, err := ContractAddressFromString(ac, addrStr)
```

Comment:
If we want to do direct string compare at `txutil`, then we need to enforce users to set this addresses as hex checksum string.
```suggestion
			contractAddr, err := ContractAddressFromString(ac, addrStr)
            if err != nil {
                return err
            } else if addrStr != contractAddr.Hex() {
                return err
            }
            
```